exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java

```diff
@@ -1,169 +1,161 @@
+            scienceQuestions.addLast(("Science Question " + index));
+            sportsQuestions.addLast(("Sports Question " + index));
+            rockQuestions.addLast(("Rock Question " + index));
+        }
+    }
+
+    public void addingPlayer(String playerName) {
+
+        players.add(playerName);
+        setUpForEachPlayer();
+        System.out.println(playerName + " was added");
+        System.out.println("They are player number " + players.size());
+    }
+
+    private void setUpForEachPlayer() {
+        placeOnTheBoard[amountOfPlayers] = startingPlaceOnTheBoard;
+        coinPurses[amountOfPlayers] = startingAmountOfCoins;
+        inPenaltyBox[amountOfPlayers] = false;
+    }
+
+
+    public void rollTheDice(int numberOnDiceAfterRoll) {
+        boolean oddNumberRolled = numberOnDiceAfterRoll % 2 != 0;
+        boolean currentPlayerIsInPenaltyBox = this.inPenaltyBox[currentPlayer];
+
+        System.out.println(players.get(currentPlayer) + " is the current player");
+        System.out.println("They have rolled a " + numberOnDiceAfterRoll);
+
+        if (currentPlayerIsInPenaltyBox) {
+            if (oddNumberRolled) {
+                isGettingOutOfPenaltyBox = true;
+                System.out.println(players.get(currentPlayer) + " is getting out of the penalty box");
+                outcomeOfCorrectAnswer(numberOnDiceAfterRoll);
+            } else {
+                System.out.println(players.get(currentPlayer) + " is not getting out of the penalty box");
+                isGettingOutOfPenaltyBox = false;
+            }
+
+        } else {
+            outcomeOfCorrectAnswer(numberOnDiceAfterRoll);
+        }
+    }
+
+    private void outcomeOfCorrectAnswer(int numberOnDiceAfterRoll) {
+        int oneMoreThanTheMaximumPlaceOnBoard = 12;
+
+        placeOnTheBoard[currentPlayer] = playersPlaceOnBoard() + numberOnDiceAfterRoll;
+
+        if (playersPlaceOnBoard() > maximumPlaceOnTheBoard) {
+            placeOnTheBoard[currentPlayer] = playersPlaceOnBoard() - oneMoreThanTheMaximumPlaceOnBoard;
+        }
+        System.out.println(players.get(currentPlayer)
+                + "'s new location is "
+                + playersPlaceOnBoard());
+        System.out.println("The category is " + currentQuestionCategory());
+        askQuestion();
+    }
+
+    private int playersPlaceOnBoard() {
+        return placeOnTheBoard[currentPlayer];
+    }
+
+    private void askQuestion() {
+        if (currentQuestionCategory().equals("Pop"))
+            System.out.println(pop.getNextQuestion());
+        if (currentQuestionCategory().equals("Science"))
+            System.out.println(scienceQuestions.removeFirst());
+        if (currentQuestionCategory().equals("Sports"))
+            System.out.println(sportsQuestions.removeFirst());
+        if (currentQuestionCategory().equals("Rock"))
+            System.out.println(rockQuestions.removeFirst());
+    }
+
+
+    private String currentQuestionCategory() {
```

Comment:
In this method there is an inconsistent way of finding out where the player is on the board, and getting the category; do you mean to define another Boolean?

Reply:
tbh im not too sure :| haha